libbpf-rs/src/query.rs

```diff
@@ -703,3 +706,39 @@ gen_info_impl!(
     libbpf_sys::bpf_link_get_next_id,
     libbpf_sys::bpf_link_get_fd_by_id
 );
+
+/// An enum describing type of eBPF object.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum BpfObjectType {
+    /// The object is a map.
+    Map,
+    /// The object is a program.
+    Program,
+    /// The object is a BPF link.
+    Link,
+}
+
+/// Get type of BPF object by fd.
+///
+/// This information is not exported directly by bpf_*_get_info_by_fd() functions,
+/// as kernel relies on the userspace code to know what kind of object it
+/// queries. The type of object can be recovered by fd only from the proc
+/// filesystem.The same approach is used in bpftool.
+pub fn object_type_from_fd(fd: BorrowedFd<'_>) -> Result<BpfObjectType> {
```

Comment:
Let's maybe not make this function public? We can use it as implementation detail for higher level APIs targetting the individual objects, but I don't think we should expose it. It just needlessly adds to the API surface, in my opinion.

Reply:
I moved it to the utils module, which is internal and doesn't expose the function publically (and updated the changelog to remove announce of that new public function).